File: weather-demo-app-android/app/src/main/java/com/demo/weather/data/WeatherDataSource.kt
package com.demo.weather.data

import com.demo.weather.domain.model.CurrentWeather
import com.demo.weather.domain.model.ForecastDay

interface WeatherDataSource {
    suspend fun getCurrentWeather(city: String): CurrentWeather
    suspend fun getForecast(city: String): List<ForecastDay>

    fun saveLastSearchedCity(city: String)
    fun getLastSearchedCity(): String?

    fun setWelcomeSeen(seen: Boolean)
    fun hasSeenWelcome(): Boolean
}

File: weather-demo-app-apple/Weather/Data/WeatherDataSource.swift
//
//  WeatherDataSource.swift
//  Weather
//
//  Created by Onur on 4/11/26.
//

protocol WeatherDataSource {
    func getCurrentWeather(city: String) async throws -> CurrentWeather
    func getForecast(city: String) async throws -> [ForecastDay]

    func saveLastSearchedCity(_ city: String)
    func getLastSearchedCity() -> String?

    func setWelcomeSeen(_ seen: Bool)
    func hasSeenWelcome() -> Bool
}
